install/0000_90_machine-config-operator_01_prometheus-rules.yaml

```diff
@@ -103,3 +103,23 @@ spec:
             severity: warning
           annotations:
             message: "Memory usage of {{ $value | humanize }} on {{ $labels.node }} exceeds 90%. Master nodes starved of memory could result in degraded performance of the control plane."
+    - name: master-nodes-info-memory-usage
+      rules:
+        - alert: MasterNodesInfoMemoryUsage
```

Comment:
MasterNodesInfoMemoryUsage doesn't seem to be a clear name?

Also the messages for all 3 alerts is the same, shouldn't they differ?

Reply:
@kikisdeliveryservice  thanks so much for your review 

Well they differ in the percentage, so we have `info` level when it crosses 40%, `warn` over 60%, and the original one should be critical as it is 90% 

I tried to use a name that shows the intentions, would you kindly suggest a better name ?

Reply:
yes i think it should be `critical` as it is 90% :) 

Also the runbooks are required for critical only afaik :)

Reply:
Ahh gotcha! Could you update that alert from warning ->critical?

Reply:
#3052 has been in flight for a while to move `HighOverallControlPlaneCPU` (`warning` >60%) and `ExtremelyHighIndividualControlPlaneCPU` (`warning` >75%, `critical` >90%) over to this repository.  Can we name-match with these, like `HighOverallControlPlaneMemory` (`warning` >60%) and `ExtremelyHighIndividualControlPlaneMemory`, or some such?  And drop or rename `MasterNodesHighMemoryUsage`?  It's less mental overhead for both devs and users if we don't have two completely separate systems for alerting on the two resource types.

> Also the runbooks are required for critical only afaik :)

The goal with alerts is to inform admins so they can do something with that information.  With `info`, maybe that's "more easily debug some other issue, because they're now aware of some related condition".  But with `warning` and `critical`, we should be very clear about:

* Why they should care?  How does >40% memory usage on a control plane node lead to degraded etcd performance?  Is it because there's less memory around for caching?  If it's degraded, can we expect the operator to go `Degraded=True`?  Explain to me so when I have 30 alerts firing from 10 clusters, I can correctly prioritize dealing with this one.

* Once they've decided they care, what should they do about it?  Is it "scale up the control plane machines" (as far as I know, still no official docs on how to accomplish this)?  It "check for large numbers of etcd resources and see if you can prune something like events or runaway pods"?  Maybe "set up some Kube resource quotas"?

We don't need a runbook to cover those points, but we should cover them in the description, and push fiddly details into a runbook if there ends up being more than we can wedge into the description annotation.

CC @openshift/sre-alert-sme